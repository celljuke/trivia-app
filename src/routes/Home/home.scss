@import './styles/_colors.scss';

.home-page {
  display: flex;
  flex-direction: column;
  justify-content: center;

    h1 {
      font-size: 4rem;
      line-height: 1.1;
      text-align: center;
      margin: 5rem 0 5rem;
    }

    p + p {
      margin-top: 2rem;
    }

    p {
      margin: 0;
      font-family: 'Monda', sans-serif;
      font-size: 2rem;
      font-weight: 400;
      line-height: 1.1;
    } 


  .begin-form {
    display: flex;
    width: 54rem;
    margin: 6rem auto;
    box-shadow: 10px 10px 0px 0px rgba(0,0,0,0.3);

    input {
      width: 34rem;
      height: 6rem;
      border: none;
      padding: 1rem 1rem 1rem 2rem;
      position: relative;
      font-family: 'Montserrat', sans-serif;
      font-size: 1.8rem;
      font-weight: 400;
      background-color: #ffffff;
      border-radius: 0;
      outline: none;
    }

    .begin-button {
      width: 20rem;
      height: 6rem;
      background-color: #e02365;
      border: none;
      color: #fff;
      font-family: 'Montserrat', sans-serif;
      font-size: 18px;
      cursor: pointer;
      border-radius: 0;
      background: linear-gradient(-45deg, #e02365, #db3c06, #f90e61, #e02365);
      outline: none;
      font-weight: 700;
    }
  }
}
