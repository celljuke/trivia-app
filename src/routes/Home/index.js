// @flow

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { RouterHistory } from 'react-router-dom';
import { setUserName, getQuestions } from '../../actions/actionCreators';
import { FlexDiv, HighLightedText } from '../../components';

import './home.css';

class Home extends Component {

  props: {
    username: string,
    handleUserNameChange: Function,
    getQuestions: Function,
    history: RouterHistory
  };

  goToQuestions = event => {
    event.preventDefault();

    if (this.props.username) {
      this.props.getQuestions()
        .then(() => this.props.history.push('/questions'));
    }
  }

  render() {
    return (
      <div className="home-page">
        <FlexDiv align="center" direction="column" justify="center">
          <h1 className="page-title">WELCOME TO THE TRIVIA CHALLENGE</h1>
          <p>
            You will be presented with <HighLightedText>10</HighLightedText>
            <HighLightedText backgroundColor="#00ca5b">True</HighLightedText>
            or
            <HighLightedText backgroundColor="#e60000">False</HighLightedText>
            questions.
          </p>
          <p>Can you score 100%?</p>
        </FlexDiv>
        <div className="begin-form">
          <form onSubmit={this.goToQuestions}>
            <input
              type="text"
              placeholder="username"
              value={this.props.username}
              onChange={this.props.handleUserNameChange} />
            <input type="button" className="begin-button" onClick={this.goToQuestions} value="BEGIN"/>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  username: state.username
});

const mapDispatchToProps = (dispatch: Function) => ({
  handleUserNameChange(event) {
    dispatch(setUserName(event.target.value));
  },
  getQuestions() {
    return dispatch(getQuestions());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
