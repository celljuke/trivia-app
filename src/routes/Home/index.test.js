import React from 'react';
import { shallow, mount } from 'enzyme';
import Home from './index';
import store from '../../store';

describe('<Home />', () => {
  it('renders without crashing', () => {
    const component = shallow(<Home store={store} />);
    expect(component).toMatchSnapshot();
  });

  it('has welcome title', () => {
    const wrapper = mount(<Home store={store} />);
    const welcome = 'WELCOME TO THE TRIVIA CHALLENGE';
    expect(wrapper.find('.page-title')).toHaveText(welcome);
  });
});
