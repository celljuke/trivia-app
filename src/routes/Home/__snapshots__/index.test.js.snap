// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders without crashing 1`] = `
<Home
  getQuestions={[Function]}
  handleUserNameChange={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  username=""
/>
`;
